cider/exec/template/StringOpsIR.cpp

```diff
@@ -270,6 +319,66 @@ llvm::Value* CodeGenerator::codegenPerRowStringOper(const Analyzer::StringOper*
                                        string_oper_lvs);
 }
 
+std::unique_ptr<CodegenColValues> CodeGenerator::codegenStringOpExpr(
+    const Analyzer::StringOper* expr,
+    const CompilationOptions& co) {
+  AUTOMATIC_IR_METADATA(cgen_state_);
+
+  CHECK_GE(expr->getArity(), 1UL);
+  CHECK(expr->hasNoneEncodedTextArg());
+
+  const auto& expr_ti = expr->get_type_info();
+  const auto primary_arg = remove_cast(expr->getArg(0));
+  CHECK(primary_arg->get_type_info().is_none_encoded_string());
+
+  auto primary_str = codegen(primary_arg, co, true);  // Twovalue
+  auto str_values = dynamic_cast<TwoValueColValues*>(primary_str.get());
+  CHECK(str_values);
+
+  const auto string_op_infos = getStringOpInfos(expr);
+  CHECK(string_op_infos.size());
+  const auto string_ops = getStringOps(string_op_infos);
+  const int64_t string_ops_handle = reinterpret_cast<int64_t>(string_ops);
+  auto string_ops_handle_lv = cgen_state_->llInt(string_ops_handle);
+  const auto& return_ti = expr->get_type_info();
+  if (!return_ti.is_string()) {
+    CIDER_THROW(
+        CiderCompileException,
+        "For string op, non-string type return values is not supported currently.");
+  }
+
+  const int64_t cider_string_hasher_handle =
+      reinterpret_cast<int64_t>(executor()->getCiderStringHasherHandle());
+  auto cider_string_hasher_handle_lv = cgen_state_->llInt(cider_string_hasher_handle);
+  std::string func_name = "apply_string_ops_and_encode_cider";
+
+  std::vector<llvm::Value*> string_oper_lvs{str_values->getValueAt(0),
+                                            str_values->getValueAt(1),
+                                            string_ops_handle_lv,
+                                            cider_string_hasher_handle_lv};
+  if (str_values->getNull()) {
+    func_name.append("_nullable");
+    string_oper_lvs.push_back(str_values->getNull());
+  }
+  auto id = cgen_state_->emitExternalCall(
+      func_name, get_int_type(64, cgen_state_->context_), string_oper_lvs);
+  llvm::Value* res_null = nullptr;
+  if (str_values->getNull()) {
+    res_null = cgen_state_->emitExternalCall(
+        "cider_check_string_id_is_null", get_int_type(1, cgen_state_->context_), {id});
```

Comment:
Why not directly equal to null of original column?

Reply:
make sense for most cases. I am not sure whether we have case like `str_not_null` -> `string_op` -> `null`, if have we can reserve for it.

Reply:
Maybe in above scenario, we should take it as `empty` string instead of `null`?